### PR TITLE
FEATURE/Google Gemini - Support for Chat Model, Credential & Embeddings

### PR DESCRIPTION
**Pull request by vinodkiran**
Screenshot 2023-12-15 at 20 49 49
Screenshot 2023-12-15 at 20 50 15

### CODE REVIEW

File: packages/components/nodes/chatmodels/ChatGoogleGenerativeAI/ChatGoogleGenerativeAI.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    category: string
+    description: string
+    baseClasses: string[]
+    credential: INodeParams
+    inputs: INodeParams[]
+
+    constructor() {
+        this.label = 'ChatGoogleGenerativeAI'
+        this.name = 'chatGoogleGenerativeAI'
+        this.version = 2.0
```
Comment (HenryHengZJ) [inline packages/components/nodes/chatmodels/ChatGoogleGenerativeAI/ChatGoogleGenerativeAI.ts:21]: should be version 1

File: packages/components/nodes/chatmodels/ChatGoogleGenerativeAI/ChatGoogleGenerativeAI.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        this.icon = 'gemini.png'
+        this.category = 'Chat Models'
+        this.description = 'Wrapper around Google Gemini large language models that use the Chat endpoint'
+        this.baseClasses = [this.type, ...getBaseClasses(ChatGoogleGenerativeAI)]
+        this.credential = {
+            label: 'Connect Credential',
+            name: 'credential',
+            type: 'credential',
+            credentialNames: ['googleGenerativeAI'],
+            optional: true,
```
Comment (HenryHengZJ) [inline packages/components/nodes/chatmodels/ChatGoogleGenerativeAI/ChatGoogleGenerativeAI.ts:32]: is credential optional?

File: packages/components/nodes/chatmodels/ChatGoogleGenerativeAI/ChatGoogleGenerativeAI.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        const topP = nodeData.inputs?.topP as string
+        const cache = nodeData.inputs?.cache as BaseCache
+
+        const obj = {
+            apiKey: apiKey,
+            modelName: modelName,
+            maxOutputTokens: 2048
+        }
+
+        if (maxOutputTokens) obj.maxOutputTokens = parseInt(maxOutputTokens, 10)
```
Comment (HenryHengZJ) [inline packages/components/nodes/chatmodels/ChatGoogleGenerativeAI/ChatGoogleGenerativeAI.ts:98]: think we can standardize and use `parseFloat`

File: packages/components/nodes/embeddings/GoogleGenerativeAIEmbedding/GoogleGenerativeAIEmbedding.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        this.icon = 'gemini.png'
+        this.category = 'Embeddings'
+        this.description = 'Google Generative API to generate embeddings for a given text'
+        this.baseClasses = [this.type, ...getBaseClasses(GoogleVertexAIEmbeddings)]
+        this.credential = {
+            label: 'Connect Credential',
+            name: 'credential',
+            type: 'credential',
+            credentialNames: ['googleGenerativeAI'],
+            optional: true,
```
Comment (HenryHengZJ) [inline packages/components/nodes/embeddings/GoogleGenerativeAIEmbedding/GoogleGenerativeAIEmbedding.ts:33]: optional credential

File: packages/components/nodes/chatmodels/ChatGoogleGenerativeAI/ChatGoogleGenerativeAI.ts (hunk 1 above)
Comment (vinodkiran) [reply packages/components/nodes/chatmodels/ChatGoogleGenerativeAI/ChatGoogleGenerativeAI.ts]: Fixed!

File: packages/components/nodes/chatmodels/ChatGoogleGenerativeAI/ChatGoogleGenerativeAI.ts (hunk 2 above)
Comment (vinodkiran) [reply packages/components/nodes/chatmodels/ChatGoogleGenerativeAI/ChatGoogleGenerativeAI.ts]: Oversight! corrected & Fixed.

File: packages/components/nodes/chatmodels/ChatGoogleGenerativeAI/ChatGoogleGenerativeAI.ts (hunk 3 above)
Comment (vinodkiran) [reply packages/components/nodes/chatmodels/ChatGoogleGenerativeAI/ChatGoogleGenerativeAI.ts:98]: Tokens are usually parseInt. But changed temperature to parseFloat and pushed the changes.

File: packages/components/nodes/embeddings/GoogleGenerativeAIEmbedding/GoogleGenerativeAIEmbedding.ts (hunk 4 above)
Comment (vinodkiran) [reply packages/components/nodes/embeddings/GoogleGenerativeAIEmbedding/GoogleGenerativeAIEmbedding.ts]: Oversight! corrected & Fixed.

**Review by HenryHengZJ — APPROVED** [review]

**Pull request merged**